### PR TITLE
Add docker compose support for --build and --rmi options

### PR DESCRIPTION
**Pull request by rnorth**
Replaces #1336
(credit to @sonerd)

> Took the changes made here #924 and added test for docker-compose --build

### CODE REVIEW

File: core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package org.testcontainers.junit;
+
+import java.io.File;
+import org.junit.Rule;
+import org.testcontainers.containers.DockerComposeContainer;
+
+public class DockerComposeContainerWithBuildTest extends BaseDockerComposeTest {
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java]: I am afraid this test does not test it, and only covers the setter. If we break `withBuild`, the test will continue to pass.

Also, the used compose file (https://github.com/testcontainers/testcontainers-java/blob/151b93ac065a420d0a803d2ebb8ba6d152c783a4/core/src/test/resources/compose-test.yml ) is rather heavy (mysql)

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java [hunk 2]
```diff
@@ -465,6 +497,18 @@ private SELF self() {
     private String randomProjectId() {
         return identifier + Base58.randomString(6).toLowerCase();
     }
+
+    public enum RemoveImages {
+        /**
+         * Remove all images used by any service.
+         */
+        all,
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: perhaps we should use standard naming convention for enums (upper case) and have a field with flag's value?

File: core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java (hunk 1 above)
Comment (sonerd) [reply core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java]: Regarding the test I was working on a version which tests the created image and so on. 
Due to time issues I was not bale to finish it.

I can use a lighter compose file.

File: core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java (hunk 2 above)
Comment (sonerd) [reply core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: Sure, can  do it.

Comment (rnorth) [reply core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java]: I've taken this on and am pushing the proposed change now.

File: core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java (hunk 1 above)
Comment (rnorth) [reply core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java]: I've taken this on and am pushing the proposed change now.

File: core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        doTest(LOCAL, false, true);
+    }
+
+
+    @Test
+    public void allModeRemovesAllImages() {
+        doTest(ALL, false, false);
+    }
+
+    private void doTest(final DockerComposeContainer.RemoveImages removeMode,
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java]: @rnorth WDYT about using the parameterized runner instead?

Comment (rnorth) [reply core/src/test/java/org/testcontainers/junit/DockerComposeContainerWithBuildTest.java]: May as well - done!

**Pull request merged**